framework/sandstone.cpp

```diff
@@ -3801,10 +3805,17 @@ int main(int argc, char **argv)
             logging_print_iteration_start();
             initialize_smi_counts();  // used by smi_count test
         } else if (lastTestResult != TestResult::Skipped) {
-            if (sApp->service_background_scan)
-                background_scan_wait();
-            else
-                wait_delay_between_tests();
+            if (sApp->service_background_scan) {
+                if (!background_scan_wait()) {
+                    logging_printf(LOG_LEVEL_VERBOSE(2), "# Background scan: waiting between tests interrupted\n");
+                    break;
+                }
+            } else {
+                if (!wait_delay_between_tests()) {
+                    logging_printf(LOG_LEVEL_VERBOSE(2), "# Test execution interrupted between tests\n");
+                    break;
+                }
```

Comment:
It feels that this one should do an `exit: interrupted`, maybe both. It's a matter of chance whether you hit the sleep between tests or whether you hit the test, so it shouldn't change the output.

WDYT?

Reply:
As it doesn't affect any test state, "original" report seems to be better. What more, this is almost identical with "normal" test done.
If we're expected to report "interrupted".. then all signals should be verified and detailed state ("interrupted", "terminated", "killed" with appropriate signal) should be reported.
Overall I think, that composite status of test execution would be best (at the end, e.g.: "failed/interrupted"). This would be much bigger task.

Reply:
Maybe we should arm/disarm the handlers around `wait_for_children` so if these issues happen while sleeping, the process simply exits. The YAML output will be in a good state, though it will miss the "exit:" field.

Reply:
I withdraw that. I still think we need an `exit: interrupted` because we did not run all tests we were planning to. But I can also see how someone using `-T forever` would want to get the result of tests that were run.